### ISSUE TITLE
Agent Tools dialog: Change in content from buttons

### ISSUE THREAD

**Issue by kangabell** [danny-avila/LibreChat#5505]
When adding a new agent tool that requires an API key, a new form to add the key appears but is not announced to the screen reader. To a non-sighted user, these buttons appear to do nothing.

## WCAG Criteria
[4.1.3 Status Messages - AA](https://www.w3.org/WAI/WCAG21/Understanding/status-messages)

## Testing Method
Manual - VoiceOver, Chrome, MacOS

## Notes
Move keyboard focus to the API key field when button is clicked, or [use aria-live to announce the change in content](https://accessibility.huit.harvard.edu/provide-notification-dynamic-changes-content).

## Screenshot or Screen Recording
[View Screen Recording](https://share.zight.com/YEu6BKr0)

**Comment by kangabell** [comment]
Originally identified in #5187:
>When you add a tool to an agent that requires to enter an API key, the input box to open that key should automatically get focus. Otherwise, it just appears on top of the tools list, and I as the user wonder why nothing happens when I press the add button.

**Comment by kangabell** [comment]
There is a similar issue with the dialog that appears after selecting "Run Code" in the Agent Builder panel under "Code Interpreter API".

**Comment by danny-avila** [comment]
FYI @dustinhealy
